crates/matrix-sdk/src/event_cache/mod.rs

```diff
@@ -582,6 +582,23 @@ impl RoomEventCache {
             warn!("couldn't save event without event id in the event cache");
         }
     }
+
+    /// Save some events in the event cache, for further retrieval with
+    /// [`Self::event`]. This function will save them using a single lock,
+    /// as opposed to [`Self::save_event`].
+    // This doesn't insert the event into the linked chunk. In the future there'll
+    // be no distinction between the linked chunk and the separate cache.
```

Comment:
Why?
Can we add a `TODO` and open an issue to track this please?

Reply:
The comment is just a copy of the one above it written by @bnjbvr since it follows the same logic, but I think it's probably because we lack the context of where the `LinkedChunk` containing events to save would go, or if it even makes sense to add them to one (these events can come from very different sources and may not be related enough to be added as a chunk of a timeline, if I understood how `LinkedChunk` works properly).

I guess the 'in the future' part means once we have a persistent event cache, but I'm not 100% sure.

Anyway,  can create an issue for that.

Reply:
@Hywan I added the TODO [here](https://github.com/matrix-org/matrix-rust-sdk/pull/3840/commits/4f57f54b7b7dfba0894aa16c5b7547a104155771) and opened the issue [here](https://github.com/matrix-org/matrix-rust-sdk/issues/3886).

Reply:
Excellent, thanks!